postgres/datadog_checks/postgres/statement_samples.py

```diff
@@ -628,6 +628,38 @@ def _row_to_raw_statement_event(self, row):
 
         self._check.database_monitoring_query_sample(json.dumps(raw_query_event, default=default_json_event_encoding))
 
+    # Expects one or more SQL statements in a string. If the string
+    # begins with any SET statements, they are removed and the rest
+    # of the string is returned. Otherwise, the string is returned
+    # as it was received.
+    def _trim_leading_set_stmts(self, obfuscated_statement):
+        match = re.match(
```

Comment:
If this is a static regex pattern, we should [compile](https://docs.python.org/3/library/re.html#re.compile) it once at module load time in order to avoid recompiling this for every call which will be extra cpu cycles.

Reply:
I'll consider adding it as a constant, though I'm also considering switching it over to explicit code…

Reply:
So, python has an internal cache for regexes, and automatically adds things as they are encountered in code; the one case this will be a problem is if the program exceeds that cache size (I think it defaults to 100). Unfortunately, I see some odd use of `compile` in loops in this codebase, so I agree an explicit `compile` makes sense in this case.